# Most Money Making Movie by Genre

Methods and Results

First of all, I did some data cleaning work on the disney_movies_total_gross dataset to convert the release date value(object) into datatype(datetime64), remove the rows with null values, assign new column and convert the currency value(object) to datatype(float). Those steps allowed me to group, sort values, calculate, and plot the graphs.

Let's check out my data cleaning step by step.

In [1]:
# Lets import all the required libraries needed for the analysis
import altair as alt
import pandas as pd

# import all the required files
gross = pd.read_csv("data/disney_movies_total_gross.csv")
directors = pd.read_csv("data/disney-director.csv")

We can read in the gross dataframe by the following python code:

```python
gross.head()
```

In [2]:
# read in the gross dataframe
gross.head()

movie_title  release_date      genre MPAA_rating  \
0  Snow White and the Seven Dwarfs  Dec 21, 1937    Musical           G   
1                        Pinocchio   Feb 9, 1940  Adventure           G   
2                         Fantasia  Nov 13, 1940    Musical           G   
3                Song of the South  Nov 12, 1946  Adventure           G   
4                       Cinderella  Feb 15, 1950      Drama           G   

    total_gross inflation_adjusted_gross  
0  $184,925,485           $5,228,953,251  
1   $84,300,000           $2,188,229,052  
2   $83,320,000           $2,187,090,808  
3   $65,000,000           $1,078,510,579  
4   $85,000,000             $920,608,730

In [3]:
# read in the directors dataframe

directors.head()

name        director
0  Snow White and the Seven Dwarfs      David Hand
1                        Pinocchio  Ben Sharpsteen
2                         Fantasia    full credits
3                            Dumbo  Ben Sharpsteen
4                            Bambi      David Hand

Let's get some other information about the directors dataset and the gross dataset by running the python code:

```python
directors.info()
```

```python
gross.info()
```

In [4]:
#check info of the directors dataframe
directors.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 56 entries, 0 to 55
Data columns (total 2 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   name      56 non-null     object
 1   director  56 non-null     object
dtypes: object(2)
memory usage: 1.0+ KB


The directors dataset has 56 rows and 2 columns. All 56 rows have non-null values.

In [5]:
#check info of the gross dataframe
gross.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 579 entries, 0 to 578
Data columns (total 6 columns):
 #   Column                    Non-Null Count  Dtype 
---  ------                    --------------  ----- 
 0   movie_title               579 non-null    object
 1   release_date              579 non-null    object
 2   genre                     562 non-null    object
 3   MPAA_rating               523 non-null    object
 4   total_gross               579 non-null    object
 5   inflation_adjusted_gross  579 non-null    object
dtypes: object(6)
memory usage: 27.3+ KB


The gross dataset has 579 rows and 6 columns. The genre and MPAA-rating columns have null values. The next step I took was to use the argument parse_dates to convert the release_date into the datetime64 and then remove the null values.

In [6]:
# parse_dates parameter to convince pandas to turn the release_date values into real datetime types
gross = pd.read_csv("data/disney_movies_total_gross.csv", parse_dates = ['release_date'])
gross

movie_title release_date      genre MPAA_rating  \
0    Snow White and the Seven Dwarfs   1937-12-21    Musical           G   
1                          Pinocchio   1940-02-09  Adventure           G   
2                           Fantasia   1940-11-13    Musical           G   
3                  Song of the South   1946-11-12  Adventure           G   
4                         Cinderella   1950-02-15      Drama           G   
..                               ...          ...        ...         ...   
574         The Light Between Oceans   2016-09-02      Drama       PG-13   
575                   Queen of Katwe   2016-09-23      Drama          PG   
576                   Doctor Strange   2016-11-04  Adventure       PG-13   
577                            Moana   2016-11-23  Adventure          PG   
578     Rogue One: A Star Wars Story   2016-12-16  Adventure       PG-13   

      total_gross inflation_adjusted_gross  
0    $184,925,485           $5,228,953,251  
1     $84,300,000           $2,188,229,052  
2     $83,320,000           $2,187,090,808  
3     $65,000,000           $1,078,510,579  
4     $85,000,000             $920,608,730  
..            ...                      ...  
574   $12,545,979              $12,545,979  
575    $8,874,389               $8,874,389  
576  $232,532,923             $232,532,923  
577  $246,082,029             $246,082,029  
578  $529,483,936             $529,483,936  

[579 rows x 6 columns]

In [7]:
# drop the rows with null values
gross = gross.dropna()
gross

movie_title release_date      genre MPAA_rating  \
0    Snow White and the Seven Dwarfs   1937-12-21    Musical           G   
1                          Pinocchio   1940-02-09  Adventure           G   
2                           Fantasia   1940-11-13    Musical           G   
3                  Song of the South   1946-11-12  Adventure           G   
4                         Cinderella   1950-02-15      Drama           G   
..                               ...          ...        ...         ...   
574         The Light Between Oceans   2016-09-02      Drama       PG-13   
575                   Queen of Katwe   2016-09-23      Drama          PG   
576                   Doctor Strange   2016-11-04  Adventure       PG-13   
577                            Moana   2016-11-23  Adventure          PG   
578     Rogue One: A Star Wars Story   2016-12-16  Adventure       PG-13   

      total_gross inflation_adjusted_gross  
0    $184,925,485           $5,228,953,251  
1     $84,300,000           $2,188,229,052  
2     $83,320,000           $2,187,090,808  
3     $65,000,000           $1,078,510,579  
4     $85,000,000             $920,608,730  
..            ...                      ...  
574   $12,545,979              $12,545,979  
575    $8,874,389               $8,874,389  
576  $232,532,923             $232,532,923  
577  $246,082,029             $246,082,029  
578  $529,483,936             $529,483,936  

[513 rows x 6 columns]

Now the gross dataset has 513 rows and 6 columns without any null values. The release_date has been converted to datetime64 but the total_gross and inflation_adjusted_gross are object, which don't help us read and operate the data correctly. So the next steps were to assign a new column for the year and convert the total_gross and inflation_adjusted_gross to float.

In [8]:
# assign a new column "year" for the release year
gross = gross.assign(year = gross['release_date'].dt.year )
gross.head()

movie_title release_date      genre MPAA_rating  \
0  Snow White and the Seven Dwarfs   1937-12-21    Musical           G   
1                        Pinocchio   1940-02-09  Adventure           G   
2                         Fantasia   1940-11-13    Musical           G   
3                Song of the South   1946-11-12  Adventure           G   
4                       Cinderella   1950-02-15      Drama           G   

    total_gross inflation_adjusted_gross  year  
0  $184,925,485           $5,228,953,251  1937  
1   $84,300,000           $2,188,229,052  1940  
2   $83,320,000           $2,187,090,808  1940  
3   $65,000,000           $1,078,510,579  1946  
4   $85,000,000             $920,608,730  1950

In [9]:
# convert the total_gross column from object to float
gross= gross.assign(total_gross = gross['total_gross'].replace( '[\$,)]','', regex=True ).astype(float))
gross.head()

movie_title release_date      genre MPAA_rating  \
0  Snow White and the Seven Dwarfs   1937-12-21    Musical           G   
1                        Pinocchio   1940-02-09  Adventure           G   
2                         Fantasia   1940-11-13    Musical           G   
3                Song of the South   1946-11-12  Adventure           G   
4                       Cinderella   1950-02-15      Drama           G   

   total_gross inflation_adjusted_gross  year  
0  184925485.0           $5,228,953,251  1937  
1   84300000.0           $2,188,229,052  1940  
2   83320000.0           $2,187,090,808  1940  
3   65000000.0           $1,078,510,579  1946  
4   85000000.0             $920,608,730  1950

In [10]:
# convert the inflation_adjusted_gross column from object to float
gross = gross.assign(inflation_adjusted_gross = gross['inflation_adjusted_gross'].replace( '[\$,)]','', regex=True ).astype(float))
gross.head()

movie_title release_date      genre MPAA_rating  \
0  Snow White and the Seven Dwarfs   1937-12-21    Musical           G   
1                        Pinocchio   1940-02-09  Adventure           G   
2                         Fantasia   1940-11-13    Musical           G   
3                Song of the South   1946-11-12  Adventure           G   
4                       Cinderella   1950-02-15      Drama           G   

   total_gross  inflation_adjusted_gross  year  
0  184925485.0              5.228953e+09  1937  
1   84300000.0              2.188229e+09  1940  
2   83320000.0              2.187091e+09  1940  
3   65000000.0              1.078511e+09  1946  
4   85000000.0              9.206087e+08  1950

Now the gross dataset is cleaned up. Let's create a function to compute the grouping and sum values of the gross.

As we can see from the above dataset, Snow White and the Seven Dwarfs {cite:p}`SnowWhite` released in 1937 is ranking the top gross revenue movie. {numref}`movie-figure` is the movie's poster.

```{figure} https://upload.wikimedia.org/wikipedia/en/4/49/Snow_White_1937_poster.png
---
height: 300px
name: movie-figure
---
Snow White and the Seven Dwarfs
```

In [11]:
# run it on the data and group the Genre and sum up the total_gross
genre_sum = gross.groupby('genre')['total_gross'].sum().sort_values(ascending=False).reset_index()
genre_sum                                       

genre   total_gross
0             Adventure  1.617846e+10
1                Comedy  7.790333e+09
2                Action  4.122489e+09
3                 Drama  4.013773e+09
4     Thriller/Suspense  1.394100e+09
5               Musical  1.139413e+09
6       Romantic Comedy  1.120583e+09
7               Western  3.590115e+08
8           Documentary  1.806856e+08
9   Concert/Performance  1.034565e+08
10         Black Comedy  9.754321e+07
11               Horror  8.141278e+07

I grouped the movie by genre and calculated the sum of the total gross of each group. Below is the sum equation. 


```{math}
:label: sum_label
x_{1} + x_{2} + x_{3} +\dots + x_{n}=\sum_{n=1}^{10}n
```

Equation {eq}`sum_label` is to calculate the sum of all values.

In [12]:
def group_count(df, group_col, group_count_col):
    group_count = df.groupby(group_col)[group_count_col].count().sort_values(ascending=False).reset_index()
    return group_count

# run the function for counting the occurences of the movie within the genre groups
genre_count = group_count(gross,'genre','movie_title')
genre_count

genre  movie_title
0                Comedy          162
1             Adventure          119
2                 Drama          103
3                Action           36
4     Thriller/Suspense           23
5       Romantic Comedy           22
6           Documentary           16
7               Musical           15
8               Western            7
9                Horror            5
10         Black Comedy            3
11  Concert/Performance            2

In [13]:
# merge the two datasets - genre_sum and genre_count to see the total gross of the genre groups and how many movies in the genre groups
genre_gross_count = genre_sum.merge(genre_count, left_on='genre', right_on='genre', how='inner').sort_values(by='total_gross',ascending=False)
genre_gross_count

genre   total_gross  movie_title
0             Adventure  1.617846e+10          119
1                Comedy  7.790333e+09          162
2                Action  4.122489e+09           36
3                 Drama  4.013773e+09          103
4     Thriller/Suspense  1.394100e+09           23
5               Musical  1.139413e+09           15
6       Romantic Comedy  1.120583e+09           22
7               Western  3.590115e+08            7
8           Documentary  1.806856e+08           16
9   Concert/Performance  1.034565e+08            2
10         Black Comedy  9.754321e+07            3
11               Horror  8.141278e+07            5

Although Comedy has 43 movies than Adventure but it earned less than the Adventure movies.

$162$$-119$ = $43$



```{math}
:label: subtract_label
a-b = c
```
Equation {eq}`subtract_label` is to calculate the difference between two values.

In [14]:
# Visualize the genre groups with the total gross box office using a circle plot


genre_gross_count_plot = (alt.Chart(genre_gross_count, width=300, height=300)
    .mark_circle(color='blue', size=300)
    .encode(
        x=alt.X("genre:N", title="Genre_Group", sort="-y"),
        y=alt.Y("total_gross:Q", title="Total_Gross_Genre"),
    )
    .properties(title="Genre_Gross")
)
genre_gross_count_plot 

C:\Users\zengm\miniconda3\lib\site-packages\altair\utils\core.py:317: FutureWarning: iteritems is deprecated and will be removed in a future version. Use .items instead.
  for col_name, dtype in df.dtypes.iteritems():


alt.Chart(...)

It is interesting to know that the "Adventure" genre has the top gross revenue.

Pinocchio {cite:p}`Pinocchio` released in 1940 is one of the top ranking Adventure movie. {numref}`pinocchio-figure` is the movie's poster.

```{figure} https://upload.wikimedia.org/wikipedia/en/b/ba/Pinocchio-1940-poster.jpg
---
height: 300px
name: pinocchio-figure
---
Pinocchio
```


In [15]:
# create a function for grouping the selected column and counting the occurences of the movie within the groups
def group_count(df, group_col, group_count_col):
    group_count = df.groupby(group_col)[group_count_col].count().sort_values(ascending=False).reset_index()
    return group_count

# run the function for counting the occurences of the movie within the year groups
year_count = group_count(gross,'year','movie_title')
year_count.head()

year  movie_title
0  1995           32
1  1996           27
2  1994           27
3  1993           24
4  1997           23

In [16]:
# Visualize the genre groups with the total gross box office using a circle plot

year_count_plot = (alt.Chart(year_count, width=500, height=500)
    .mark_bar(color='red', opacity=1)
    .encode(
        x=alt.X("year:N", title="Year_Group", sort="-y"),
        y=alt.Y("movie_title:Q", title="Total_Movie_Count"),
    )
    .properties(title="Year_Movie_Count")
)
year_count_plot 

C:\Users\zengm\miniconda3\lib\site-packages\altair\utils\core.py:317: FutureWarning: iteritems is deprecated and will be removed in a future version. Use .items instead.
  for col_name, dtype in df.dtypes.iteritems():


alt.Chart(...)